src/deepsparse/transformers/pipelines.py

```diff
@@ -0,0 +1,798 @@
+        deepsparse and onnxruntime
+    :param input_names: list of input names to the neural network
+    :param args_parser: Reference to the object in charge of parsing supplied
+        pipeline parameters. A default is provided if None
+    :param binary_output: if True, stores outputs as pickled binaries to avoid
+        storing large amount of textual data. Default is False
+    """
+
+    default_input_names = None
+
+    def __init__(
+        self,
+        model: Union[Engine, InferenceSession],
+        tokenizer: PreTrainedTokenizer,
+        config: PretrainedConfig,
+        engine: str,
+        task: str = "",
+        input_names: Optional[List[str]] = None,
+        args_parser: ArgumentHandler = None,
+        binary_output: bool = False,
+    ):
+
+        _validate_transformers_import()
+
+        self.model = model
+        self.tokenizer = tokenizer
+        self.config = config
+        self.engine = engine
+        self.input_names = input_names
+        self.binary_output = binary_output
+        self._args_parser = args_parser or DefaultArgumentHandler()
+        self._framework = (
+            "np" if self.engine in [DEEPSPARSE_ENGINE, ORT_ENGINE] else "pt"
+        )
+
+    def transform(self, X):
+        """
+        Scikit / Keras interface to transformers' pipelines.
+        This method will forward to __call__().
+        """
+        return self(X=X)
+
+    def predict(self, X):
+        """
+        Scikit / Keras interface to transformers' pipelines.
+        This method will forward to __call__().
+        """
+        return self(X=X)
+
+    def _parse_and_tokenize(
+        self, *args, padding=True, add_special_tokens=True, **kwargs
+    ):
+        # Parse arguments
+        inputs = self._args_parser(*args, **kwargs)
+        inputs = self.tokenizer(
+            inputs,
+            add_special_tokens=add_special_tokens,
+            return_tensors=self._framework,
+            padding=padding,
+        )
+
+        return inputs
+
+    def __call__(self, *args, **kwargs):
+        inputs = self._parse_and_tokenize(*args, **kwargs)
+        self._forward(inputs)
+
+    def _forward(self, inputs, return_tensors=False):
+
+        if self.engine == ORT_ENGINE:
+
+            # TODO: filter by valid name
+            #  inputs = {k: v for k, v in inputs.items() if k in self.input_names}
+            return self.model.run(None, dict(zip(self.input_names, inputs.values())))
+        elif self.engine == DEEPSPARSE_ENGINE:
+            return self.model.run(list(inputs.values()))
+        # TODO: torch
+        # with self.device_placement():
+        #         with torch.no_grad():
+        #             inputs = self.ensure_tensor_on_device(**inputs)
+        #             predictions = self.model(**inputs)[0].cpu()
+        # if return_tensors:
+        #     return predictions
+        # else:
+        #     return predictions.numpy()
```

Comment:
is this a todo for this diff?

Reply:
leaving for future work if we add torch support